construction_progress_management/data/ir_actions_server.xml

```diff
@@ -0,0 +1,61 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+    <record id="action_sol_update_progress_increment_on_change" model="ir.actions.server">
+        <field name="code"><![CDATA[
+record['x_progress_increment'] = record.x_quantity_increment / record.product_uom_qty if record.product_uom_qty > 0 else False
+]]></field>
+        <field name="model_id" ref="sale.model_sale_order_line"/>
+        <field name="state">code</field>
+        <field name="name">Construction - Update Progress Increment on Quantity Increment change</field>
+    </record>
+    <record id="action_sol_update_quantity_increment_on_change" model="ir.actions.server">
+        <field name="code"><![CDATA[
+record['x_quantity_increment'] = record.x_progress_increment * record.product_uom_qty
+]]></field>
+        <field name="model_id" ref="sale.model_sale_order_line"/>
+        <field name="state">code</field>
+        <field name="name">Construction - Update Quantity Increment on Progress Increment change</field>
+    </record>
+    <record id="action_sol_update_section_progress" model="ir.actions.server">
+        <field name="code"><![CDATA[
+lines = record.order_id.order_line.sorted(key=lambda sol: sol.sequence)
+i = [sol.sequence for sol in lines].index(record.sequence) + 1
+while (i < len(lines) and
+      ((record.display_type == 'line_section' and lines[i].display_type != 'line_section')
+      or (record.display_type == 'line_subsection' and (lines[i].display_type not in ['line_subsection', 'line_section'])))):
```

Comment:
record.display_type is and will remain 'line_section' forever, no need to check  + second condition seems to be included in third condition. Am I correct to say it is equivalent to 
```suggestion
while (i < len(lines) and lines[i].display_type not in ['line_subsection', 'line_section']):
```